xcat3/common/rest.py

```diff
@@ -0,0 +1,45 @@
+from oslo_log import log as logging
+from xcat3.common import exception
+
+import requests
+import json
+
+LOG = logging.getLogger(__name__)
+
+headers = {'Content-Type': 'application/json'}
+
+def login (url, in_data) :
+    session = requests.Session()
+    data = json.dumps(in_data)
+    print data
+    response = session.post(url, data=data, headers=headers, verify = False)
+
+    print response.text
+
+    if response.status_code != 200 :
+        raise exception.TemporaryFailure() 
+
+    return session
+
+def put (session, url, in_data) :
+    data = json.dumps(in_data)
+    response = session.put(url, data=data, headers=headers)
+
+    print response.text
+
+    if response.status_code != 200 :
+        raise exception.TemporaryFailure()
```

Comment:
You should return the correct exception and pass the correct message, so that the user or the developer can get the failure reason or traceback message.

You could define a new exception if needed.